arkouda/groupbyclass.py

```diff
@@ -1205,20 +1207,25 @@ def is_registered(self) -> bool:
 
             registered = len(simple_registered) + len(cat_registered)
             if 0 < registered < total:
-                print(f"WARNING: GroupBy {self.name} expected {total} components to be registered,"
-                      f" but only located {registered}. Found components have been unregistered.")
+                warnings.warn(f"WARNING: GroupBy {self.name} expected {total} components to be registered,"
+                              f" but only located {registered}.")
                 return False
             else:
                 return registered == total
         else:
             parts_registered: List[bool] = []
+            total = 0
             for k, v in GroupBy._get_groupby_required_pieces(self).items():
                 if k != "unique_keys" or not isinstance(self.unique_keys, Categorical):
-                    parts_registered.append(v.is_registered())
+                    reg = v.is_registered()
+                    parts_registered.append(reg)
+
+                    if reg:
+                        total += 1
 
             if any(parts_registered) and not all(parts_registered):  # test for error
-                print(f"WARNING: GroupBy {self.name} expected {len(self.keys)} components to be registered,"
-                      f" but only located {len(parts_registered)}. Found components have been unregistered.")
+                warnings.warn(f"WARNING: GroupBy {self.name} expected {len(parts_registered)} "
+                              f"components to be registered, but only located {total}.")
```

Comment:
i think you can make this slightly simpler by removing total and doing
```python
but only located {sum(parts_registered)}.
```

Reply:
Adjusted this calculation to use `sum(parts_registered)`